qutip/rcsolve.py

```diff
@@ -0,0 +1,156 @@
+#    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+#    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+#    PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+#    HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+#    SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+#    LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+#    DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+#    THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+#    (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+#    OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+###############################################################################
+"""
+This module provides exact solvers for a system-bath setup using the
+reaction coordinate method.
+"""
+
+# Author: Neill Lambert, Anubhav Vardhan
+# Contact: nwlambert@gmail.com
+
+__all__ = ['rcsolve']
+
+import warnings
+import numpy as np
+import scipy.sparse as sp
+from numpy import matrix
+from numpy import linalg
+from qutip import spre, spost, sprepost, thermal_dm, mesolve, Odeoptions
+from qutip import tensor, identity, destroy, sigmax, sigmaz, basis, qeye, dims
+
+
+def rcsolve(Hsys, psi0, tlist, e_ops, Q, wc, alpha, N, w_th, sparse=False,
+            options=None):
+    """
+    Function to solve for an open quantum system using the
+    reaction coordinate (RC) model.
+
+    Parameters
+    ----------
+    Hsys: Qobj
+        The system hamiltonian.
+    psi0: Qobj
+        Initial state of the system.
+    tlist: List.
+        Time over which system evolves.
+    e_ops: list of :class:`qutip.Qobj` / callback function single
+        Single operator or list of operators for which to evaluate
+        expectation values.
+    Q: Qobj
+        The coupling between system and bath.
+    wc: Float
+        Cutoff frequency.
+    alpha: Float
+        Coupling strength.
+    N: Integer
+        Number of cavity fock states.
+    w_th: Float
+        Temperature.
+    sparse: Boolean
+        Optional argument to call the sparse eigenstates solver if needed.
+    options : :class:`qutip.Options`
+        With options for the solver.
+
+    Returns
+    -------
+    output: Result
+        System evolution.
+    """
+    if options is None:
+        options = Options()
+
+    dot_energy, dot_state = Hsys.eigenstates(sparse=sparse)
+    deltaE = dot_energy[1] - dot_energy[0]
+    if (w_th < deltaE/2):
+        warnings.warn("Given w_th might not provide accurate results")
+    gamma = deltaE / (2 * np.pi * wc)
+    wa = 2 * np.pi * gamma * wc  # reaction coordinate frequency
+    g = np.sqrt(np.pi * wa * alpha / 2.0)  # reaction coordinate coupling
+    nb = (1 / (np.exp(wa/w_th) - 1))
+
+    # Reaction coordinate hamiltonian/operators
+    Nmax = N * 2
```

Comment:
minor reminder for me and anubhav;  this should be    Nmax = N * dimensions[1][0]      and should appear below the definition of dimensions